Make WKTextTouchBarItemController::_webViewImpl a WeakPtr

fred-wang opened the pull request:
#### 27ad881c7434f0afe6a4ac29c0c1ae6577818a53
<pre>
Make WKTextTouchBarItemController::_webViewImpl a WeakPtr
<a href="https://bugs.webkit.org/show_bug.cgi?id=295695">https://bugs.webkit.org/show_bug.cgi?id=295695</a>

Reviewed by Anne van Kesteren.

All equivalent interfaces with member pointing to a WebKit::WebViewImpl
already have it as WeakPtr. This also fixes the remaining
NoUncheckedPtrMemberChecker failure in our code.

* Source/WebKit/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations: Removed remaining failure.
* Source/WebKit/UIProcess/mac/WebViewImpl.mm: Make _webViewImpl a WeakPtr.
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKTextTouchBarItemController textListViewController]): Get raw pointer from _webViewImpl.

Canonical link: <a href="https://commits.webkit.org/297209@main">https://commits.webkit.org/297209@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb56ab6876e072cd4065153e777746c4b6b59267

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110930 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30593 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21028 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116961 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/61198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31273 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39179 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84344 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/61198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113878 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24990 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99881 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64788 "Found 1 new API test failure: /WPE/TestMultiprocess:/webkit/WebKitWebView/multiprocess-create-ready-close (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24340 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60759 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94378 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18084 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119762 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37975 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28223 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/93302 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38351 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96154 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93126 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23723 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38178 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15919 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33961 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37868 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43339 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37531 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40867 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39236 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
r+ modulo EWS and assuming a rebase will happen once the first commit lands.

fred-wang committed:
Make WKTextTouchBarItemController::_webViewImpl a WeakPtr
https://bugs.webkit.org/show_bug.cgi?id=295695

Reviewed by Anne van Kesteren.

All equivalent interfaces with member pointing to a WebKit::WebViewImpl
already have it as WeakPtr. This also fixes the remaining
NoUncheckedPtrMemberChecker failure in our code.

* Source/WebKit/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations: Removed remaining failure.
* Source/WebKit/UIProcess/mac/WebViewImpl.mm: Make _webViewImpl a WeakPtr.
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKTextTouchBarItemController textListViewController]): Get raw pointer from _webViewImpl.

Canonical link: https://commits.webkit.org/297209@main
